### PR TITLE
Tvlatas/change test job triggering

### PR DESCRIPTION
**Pull request by agv426**
# Description

This is the current POC for changing up how we trigger test jobs so that we are using exactly the GIT commit revision that we need to use for master/release builds.

- This passes the full git commit hash that is to be used in a parameter to each job stage
- Each job stage explicitly uses that git commit hash in the SCM checkout
- The jobs verify and fail fast if the git commit specified is not what got checked out (it should be, but it tests it anyway)
- The primary build job automatically will trigger the tests for master and release* branches
- The primary build job has a TRIGGER_FULL_TESTS parameter to allow non-master/release* branches to trigger the tests
- It is possible to trigger testing of a specific commit from the triggering job, but you must supply the git commit to use (or NONE if you want the current tip of your branch to be used (ie: intention is that we can still do special limited runs if needed without requiring the initial full build/test job to be re-run).

# Checklist 

As the author of this PR, I have:

- [X] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by markxnelson — DISMISSED** [review]

File: ci/JenkinsfileBuildTrigger [hunk 1]
```diff
@@ -18,6 +18,10 @@ pipeline {
                         defaultValue: 'master',
                         description: 'Branch to clone and checkout the Verrazzano repo',
                         trim: true)
+        string (name: 'VERRAZZANO_OPERATOR_IMAGE',
```
Comment (agv426) [inline ci/JenkinsfileBuildTrigger:21]: I think changes in this file are old cruft not really related

File: Jenkinsfile [hunk 2]
```diff
@@ -78,7 +79,10 @@ pipeline {
                 """
 
                 script {
-                    checkout scm
+                    def scmInfo = checkout scm
+                    env.GIT_COMMIT = scmInfo.GIT_COMMIT
```
Comment (gilbode) [inline Jenkinsfile:83]: do you really need to do this?  i think the plugin sets the environment variable.

File: ci/JenkinsfileBuildTrigger [hunk 3]
```diff
@@ -33,6 +37,7 @@ pipeline {
                             parameters: [
                                 string(name: 'VERRAZZANO_BRANCH', value: params.VERRAZZANO_BRANCH),
                                 string(name: 'KUBERNETES_CLUSTER_VERSION', value: kversion)
+                                string(name: 'VERRAZZANO_OPERATOR_IMAGE', value: params.VERRAZZANO_OPERATOR_IMAGE)
```
Comment (gilbode) [inline ci/JenkinsfileBuildTrigger:40]: What job is think Jenkinsfile used for and how is it different from JenkinsfileTestTrigger?  And why does this one take VERRAZZANO_OPERATOR_IMAGE as a parameter whereas the other one takes GIT_COMMIT_TO_USE as a parameter?

**Review by gilbode — APPROVED** [review]

File: Jenkinsfile (hunk 2 above)
Comment (agv426) [reply Jenkinsfile:83]: I was having a hard time finding exactly what it gives/doesn't give us. I did however find other folks were not getting them set automatically so I did it explicitly as noted here:

https://issues.jenkins.io/browse/JENKINS-35230?focusedCommentId=333307&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-333307

File: ci/JenkinsfileBuildTrigger (hunk 3 above)
Comment (agv426) [reply ci/JenkinsfileBuildTrigger:40]: The changes in this file are cruft, this jenkinsfile will be going away once we have the new approach in place

File: ci/JenkinsfileTestTrigger [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                    VERRAZZANO_DEV_VERSION = props['verrazzano-development-version']
+                    TIMESTAMP = sh(returnStdout: true, script: "date +%Y%m%d%H%M%S").trim()
+                    SHORT_COMMIT_HASH = sh(returnStdout: true, script: "git rev-parse --short HEAD").trim()
+                    // update the description with some meaningful info
+                    currentBuild.description = SHORT_COMMIT_HASH + " : " + env.GIT_COMMIT + " : " + params.GIT_COMMIT_TO_USE
+                }
+            }
+        }
+
+        stage ('Kick off parallel tests') {
```
Comment (bhatpmk) [inline ci/JenkinsfileTestTrigger:94]: Can you please trigger job "verrazzano-uninstall-test/master" as well ?

Comment (agv426) [reply ci/JenkinsfileTestTrigger:94]: If folks are fine with the POC as it stands, I can start adding other tests in as well, we were going to incrementally add more tests in here.

Comment (bhatpmk) [reply ci/JenkinsfileTestTrigger:94]: We can add verrazzano-uninstall-test/master in a subsequent change.

Comment (agv426) [reply ci/JenkinsfileTestTrigger:94]: I think we can get this in first, watch it for a while to be sure it is stable, then add more tests into the trigger and get rid of the other job as incremental followups

**Review by gilbode — APPROVED** [review]

**Pull request merged**